pkg/flow/internal/controller/loader.go

```diff
@@ -252,9 +279,12 @@ func (l *Loader) Cleanup(stopWorkerPool bool) {
 }
 
 // loadNewGraph creates a new graph from the provided blocks and validates it.
-func (l *Loader) loadNewGraph(args map[string]any, componentBlocks []*ast.BlockStmt, configBlocks []*ast.BlockStmt) (dag.Graph, diag.Diagnostics) {
+func (l *Loader) loadNewGraph(args map[string]any, componentBlocks []*ast.BlockStmt, configBlocks []*ast.BlockStmt, declares []*Declare) (dag.Graph, diag.Diagnostics) {
 	var g dag.Graph
 
+	// Reset the module cache before loading a new graph. This step is crucial to ensure that references to outdated modules, not included in the new configuration, are removed.
+	l.moduleReferences = make(map[string][]ModuleReference)
```

Comment:
The comment says "module cache", while the field is called `moduleReferences`. Let's make sure we get the naming consistent and update the comment when needed. I would also like to know what's the purpose of this map.

Reply:
It's now handled by the component_node_manager as customComponentDependencies. It's being initialized by FindCustomComponentDependencies (// FindCustomComponentDependencies traverses the AST of the provided declare and collects references to known custom components.). I can also add a comment on the field if needed